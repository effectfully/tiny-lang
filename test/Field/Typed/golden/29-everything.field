let ?a = T;
let ?b = F;
let #v = {T, F, T};
let ?or = ?a or ?b;
let ?and = ?or or ?b;
let ?xor = ?and or ?or;
let f = 0;
let g = 1;
let h = 2;
let ?feq = g == f;
let ?fle = h <= f;
let ?flt = f < g;
let ?fge = g >= h;
let ?fgt = h > f;
let add = f + g;
let sub = g - h;
let mul = h * f;
let div = f / g;
let ?bat = #v[f];
let ?not = not ?a;
let ?neq0 = neq0 f;
let neg' = neg g;
let inv' = inv h;
let #unpack = unpack h;
let ?let = let i = 3; T;
let ?assert = assert T; T;
let ?for = for j = 0 to 2 do end; T;
let ?if = if T then T else T;
T