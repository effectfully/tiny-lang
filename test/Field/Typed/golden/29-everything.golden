SomeOf Bool (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?a_0}) (EConst (UniConst Bool True))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?b_1}) (EConst (UniConst Bool False))) (EStatement (ELet (UniVar {_uniVarUni = Vector, _uniVarVar = #v_2}) (EConst (UniConst Vector [True,False,True]))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?or_3}) (EAppBinOp Or (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?a_0})) (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?b_1})))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?and_4}) (EAppBinOp Or (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?or_3})) (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?b_1})))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?xor_5}) (EAppBinOp Or (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?and_4})) (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?or_3})))) (EStatement (ELet (UniVar {_uniVarUni = Field, _uniVarVar = f_6}) (EConst 0)) (EStatement (ELet (UniVar {_uniVarUni = Field, _uniVarVar = g_7}) (EConst 1)) (EStatement (ELet (UniVar {_uniVarUni = Field, _uniVarVar = h_8}) (EConst 2)) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?feq_9}) (EAppBinOp FEq (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_7})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_6})))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?fle_10}) (EAppBinOp FLe (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_8})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_6})))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?flt_11}) (EAppBinOp FLt (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_6})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_7})))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?fge_12}) (EAppBinOp FGe (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_7})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_8})))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?fgt_13}) (EAppBinOp FGt (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_8})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_6})))) (EStatement (ELet (UniVar {_uniVarUni = Field, _uniVarVar = add_14}) (EAppBinOp Add (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_6})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_7})))) (EStatement (ELet (UniVar {_uniVarUni = Field, _uniVarVar = sub_15}) (EAppBinOp Sub (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_7})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_8})))) (EStatement (ELet (UniVar {_uniVarUni = Field, _uniVarVar = mul_16}) (EAppBinOp Mul (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_8})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_6})))) (EStatement (ELet (UniVar {_uniVarUni = Field, _uniVarVar = div_17}) (EAppBinOp Div (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_6})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_7})))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?bat_18}) (EAppBinOp BAt (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_6})) (EVar (UniVar {_uniVarUni = Vector, _uniVarVar = #v_2})))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?not_19}) (EAppUnOp Not (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?a_0})))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?neq0_20}) (EAppUnOp Neq0 (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_6})))) (EStatement (ELet (UniVar {_uniVarUni = Field, _uniVarVar = neg'_21}) (EAppUnOp Neg (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_7})))) (EStatement (ELet (UniVar {_uniVarUni = Field, _uniVarVar = inv'_22}) (EAppUnOp Inv (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_8})))) (EStatement (ELet (UniVar {_uniVarUni = Vector, _uniVarVar = #unpack_23}) (EAppUnOp Unp (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_8})))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?let_24}) (EStatement (ELet (UniVar {_uniVarUni = Field, _uniVarVar = i_25}) (EConst 3)) (EConst (UniConst Bool True)))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?assert_26}) (EStatement (EAssert (EConst (UniConst Bool True))) (EConst (UniConst Bool True)))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?for_27}) (EConst (UniConst Bool True))) (EStatement (ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?if_29}) (EIf (EConst (UniConst Bool True)) (EConst (UniConst Bool True)) (EConst (UniConst Bool True)))) (EConst (UniConst Bool True))))))))))))))))))))))))))))))